cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -39,6 +39,11 @@
 
 namespace arrow {
 namespace fs {
+/// Custom comparison for FileInfo, we need this to use complex googletest matchers.
```

Comment:
I'm not sure that would solve your issue, but you can use `SortInfos` to get a deterministic comparand (see `arrow/filesystem/test_util.h`).

Reply:
I think the problem would remain `::testing::ElementsAreArray` also requires `operator==`.

Reply:
Fair enough.